python/hail/genetics/tests.py

```diff
@@ -90,18 +90,87 @@ def test_classes(self):
+        self.assertEqual(c_het_phased.alleles, [1, 0])
+        self.assertEqual(c_het_phased.ploidy, 2)
+        self.assertTrue(c_het_phased.phased)
+        self.assertFalse(c_het_phased.is_haploid())
+        self.assertTrue(c_het_phased.is_diploid())
+        self.assertEqual(c_het_phased.num_alt_alleles(), 1)
+        self.assertTrue(c_het_phased.one_hot_alleles(2) == [1, 1])
+        self.assertFalse(c_het_phased.is_hom_ref())
+        self.assertTrue(c_het_phased.is_het())
+        self.assertFalse(c_het_phased.is_hom_var())
+        self.assertTrue(c_het_phased.is_non_ref())
+        self.assertFalse(c_het_phased.is_het_non_ref())
+        self.assertTrue(c_het_phased.is_het_ref())
+
+        c_hom_var = Call([1, 1])
+        self.assertEqual(c_hom_var.alleles, [1, 1])
+        self.assertEqual(c_hom_var.ploidy, 2)
+        self.assertFalse(c_hom_var.phased)
+        self.assertFalse(c_hom_var.is_haploid())
+        self.assertTrue(c_hom_var.is_diploid())
+        self.assertEqual(c_hom_var.num_alt_alleles(), 2)
+        self.assertTrue(c_hom_var.one_hot_alleles(2) == [0, 2])
+        self.assertFalse(c_hom_var.is_hom_ref())
+        self.assertFalse(c_hom_var.is_het())
+        self.assertTrue(c_hom_var.is_hom_var())
+        self.assertTrue(c_hom_var.is_non_ref())
+        self.assertFalse(c_hom_var.is_het_non_ref())
+        self.assertFalse(c_hom_var.is_het_ref())
+        self.assertTrue(c_hom_var.unphased_diploid_gt_index() == 2)
+        
+        c_haploid = Call([2], phased=True)
+        self.assertEqual(c_haploid.alleles, [2])
+        self.assertEqual(c_haploid.ploidy, 1)
+        self.assertTrue(c_haploid.phased)
+        self.assertTrue(c_haploid.is_haploid())
+        self.assertFalse(c_haploid.is_diploid())
+        self.assertEqual(c_haploid.num_alt_alleles(), 1)
+        self.assertTrue(c_haploid.one_hot_alleles(3) == [0, 0, 1])
+        self.assertFalse(c_haploid.is_hom_ref())
+        self.assertFalse(c_haploid.is_het())
+        self.assertTrue(c_haploid.is_hom_var())
+        self.assertTrue(c_haploid.is_non_ref())
+        self.assertFalse(c_haploid.is_het_non_ref())
+        self.assertFalse(c_haploid.is_het_ref())
+        
+        c_zeroploid = Call([])
+        self.assertEqual(c_zeroploid.alleles, [])
+        self.assertEqual(c_zeroploid.ploidy, 0)
+        self.assertFalse(c_zeroploid.phased)
+        self.assertFalse(c_zeroploid.is_haploid())
+        self.assertFalse(c_zeroploid.is_diploid())
+        self.assertEqual(c_zeroploid.num_alt_alleles(), 0)
+        self.assertTrue(c_zeroploid.one_hot_alleles(3) == [0, 0, 0])
+        self.assertFalse(c_zeroploid.is_hom_ref())
+        self.assertFalse(c_zeroploid.is_het())
+        self.assertFalse(c_zeroploid.is_hom_var())
+        self.assertFalse(c_zeroploid.is_non_ref())
+        self.assertFalse(c_zeroploid.is_het_non_ref())
+        self.assertFalse(c_zeroploid.is_het_ref())
+
+        self.assertRaisesRegexp(NotImplementedError,
```

Comment:
raises Regexp?  Hmm.

Reply:
What's wrong with this? This is the same as TestUtils.interceptFatal in Scala.